### ISSUE TITLE
Improvement suggestion: replace `showdown` with smaller, newer library

### ISSUE THREAD

**Issue by sgomes** [WordPress/gutenberg#21972]
**Is your feature request related to a problem? Please describe.**
`showdown` is a relatively large library (74KB, 24KB gzipped), that appears to no longer be in development (last activity in the repo/`npm` dates to early November last year).

**Describe the solution you'd like**
Replacing with a smaller and actively maintained library would work towards making the `blocks` chunk smaller.

My suggestion would be `marked`, which has several benefits:
- It's only 28KB (9KB gzipped), which is significantly smaller than `showdown`
- It's actively supported and [strongly preferred by the community](https://www.npmtrends.com/markdown-it-vs-marked-vs-showdown)
- It's significantly faster at runtime than `showdown`, at around 6x the speed

**Describe alternatives you've considered**
The obvious alternative is to keep using the same library, but make the import smaller somehow. This was partially achieved in #21862, by upgrading to the latest version of `showdown` (which happened to be smaller).

Beyond that, the only option is to rely on tree-shaking to reduce the amount of the library that's being pulled in, by sticking to only what's necessary for our usage. Unfortunately, this is not possible with `showdown`, as it was coded as a monolithic single export that we have no choice but to import in its entirety. I [created an issue for an architecture revamp](https://github.com/showdownjs/showdown/issues/790) in the library repo, but have obtained no response yet, though at time of writing it has only been a couple of days.

**Comment by sgomes** [comment]
To add to this, [`remark`](https://github.com/remarkjs/remark) also looks like a good option.

It's even [more popular than `marked` at the moment](https://www.npmtrends.com/markdown-it-vs-marked-vs-remark-vs-remark-parse-vs-showdown).

It sits at around 39KB / 14KB gzipped.

**Comment by simison** [comment]
Furthermore, if the markdown renderer in Gutenberg could be exposed externally, 3rd party blocks like Jetpack Markdown could re-use it without needing to bundle their own.

https://github.com/Automattic/jetpack/blob/974d71347d082b9e9f98175b93a00d960d4220fc/extensions/blocks/markdown/renderer.js#L5

**Comment by simison** [comment]
@sgomes important to check what format of Markdown these libraries support as some of them are more extensive than others and affect quite a bit of the usefulness.

**Comment by sgomes** [comment]
> @sgomes important to check what format of Markdown these libraries support as some of them are more extensive than others and affect quite a bit of the usefulness.

@simison: Yup, absolutely. Most libraries support multiple flavours of `markdown`, but I suspect there may be different subsets even within each flavour. Is there an authoritative list of what Gutenberg expects, markdown-wise?

**Comment by Mamaduka** [comment]
Decided to give this another try - #77953.

**Comment by Mamaduka** [comment]
Closing as https://github.com/WordPress/gutenberg/pull/77953 since merge now.
